Tasks/DockerV2/dockerpush.ts

```diff
@@ -79,3 +99,58 @@ export function run(connection: ContainerConnection, outputUpdate: (data: string
 
     return promise;
 }
+
+async function publishToImageMetadataStore(connection: ContainerConnection, imageName: string, tags: string[], digest: string, dockerFilePath: string): Promise<any> {
+    const imageUri: string = getResourceName(imageName, digest);
+    const dockerFileContent: string = fs.readFileSync(dockerFilePath, 'utf-8').toString();
+    const baseImageName = getBaseImageName(dockerFileContent);
+    const layers = dockerCommandUtils.getLayers(connection, imageUri);
+    const buildId = parseInt(tl.getVariable("Build.BuildId"));
+    const buildDefinitionName = tl.getVariable("Build.DefinitionName");
+    const buildVersion = tl.getVariable("Build.BuildNumber");
+    const buildDefinitionId = tl.getVariable("System.DefinitionId");
+
+    const requestUrl: string = tl.getVariable("System.TeamFoundationCollectionUri") + "/" + tl.getVariable("System.TeamProject") + "/_apis/deployment/imagedetails?api-version=5.0-preview.1";
+    const requestBody: string = JSON.stringify(
+        {
+            "imageName": imageUri,
+            "imageUri": imageUri,
+            "hash": digest,
+            "baseImageName": baseImageName,
+            "distance": 0,
+            "imageType": "",
+            "mediaType": "",
+            "tags": tags,
+            "layerInfo": layers,
+            "buildId": buildId,
+            "buildVersion": buildVersion,
+            "buildDefinitionName": buildDefinitionName,
+            "buildDefinitionId": buildDefinitionId
+        }
+    );
+
+    const request = new WebRequest();
+    request.uri = requestUrl;
```

Comment:
Dont you need the accessToken here?

Reply:
Added the access token from 'SYSTEMVSSCONNECTION'